From a3681a59504ae7c8a17c064d513252143b77faa8 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Sat, 1 Jun 2024 00:33:56 +0200
Subject: [PATCH] Fix Systemdialog layout

Remove the outer item, it only complicates things and causes the
layouting to break

BUG: 480410
---
 components/dialogs/examples/test.qml          |   3 +-
 .../contents/systemdialog/SystemDialog.qml    | 132 ++++++++----------
 2 files changed, 61 insertions(+), 74 deletions(-)

diff --git a/components/dialogs/examples/test.qml b/components/dialogs/examples/test.qml
index 7a58fe65d95..52dd130c418 100644
--- a/components/dialogs/examples/test.qml
+++ b/components/dialogs/examples/test.qml
@@ -34,7 +34,7 @@ Kirigami.AbstractApplicationWindow {
 
         ListView {
             Layout.fillWidth: true
-            implicitHeight: 300
+            implicitHeight: 100
 
             model: ListModel {
                 ListElement {
@@ -54,6 +54,7 @@ Kirigami.AbstractApplicationWindow {
                 icon.name: "kate"
                 text: display
                 checkable: true
+                width: ListView.view.width
             }
         }
 
diff --git a/lookandfeel/org.kde.breeze/contents/systemdialog/SystemDialog.qml b/lookandfeel/org.kde.breeze/contents/systemdialog/SystemDialog.qml
index 86fa0851135..4800679be83 100644
--- a/lookandfeel/org.kde.breeze/contents/systemdialog/SystemDialog.qml
+++ b/lookandfeel/org.kde.breeze/contents/systemdialog/SystemDialog.qml
@@ -9,13 +9,12 @@ import QtQuick
 import QtQuick.Controls as QQC2
 import QtQuick.Layouts
 import QtQuick.Templates as T
-import Qt5Compat.GraphicalEffects
 import org.kde.kirigami as Kirigami
 
 /**
  * Dialog used on desktop. Uses SSDs (as opposed to CSDs).
  */
-Item {
+ColumnLayout {
     id: root
 
     property alias mainItem: contentsControl.contentItem
@@ -26,102 +25,89 @@ Item {
     readonly property alias dialogButtonBox: footerButtonBox
 
     property Window window
-    implicitHeight: column.implicitHeight
-    implicitWidth: column.implicitWidth
-    readonly property real minimumHeight: column.Layout.minimumHeight + (mainItem ? mainItem.implicitHeight : 0) + footerButtonBox.implicitHeight + root.spacing * 2
-    readonly property real minimumWidth: Math.min(Math.round(Screen.width / 3), column.Layout.minimumWidth + (mainItem ? mainItem.implicitWidth : 0) + footerButtonBox.implicitWidth) + root.spacing * 2
     readonly property int flags: Qt.Dialog | Qt.WindowCloseButtonHint | Qt.WindowTitleHint | Qt.WindowSystemMenuHint
     property alias standardButtons: footerButtonBox.standardButtons
-    readonly property int spacing: Kirigami.Units.largeSpacing // standard KDE dialog margins
+
+    readonly property real minimumHeight: implicitHeight
+    readonly property real minimumWidth: Math.min(Math.round(Screen.width / 3), implicitWidth)
 
     function present() {
-        window.show()
+        window.show();
     }
 
+    spacing: Kirigami.Units.smallSpacing
+
     Kirigami.Separator {
-        anchors {
-            top: parent.top
-            left: parent.left
-            right: parent.right
-        }
+        Layout.fillWidth: true
     }
 
-    ColumnLayout {
-        id: column
-        anchors.fill: parent
-        anchors.margins: root.spacing
+    RowLayout {
 
-        spacing: root.spacing
+        Layout.leftMargin: Kirigami.Units.largeSpacing
+        Layout.rightMargin: Kirigami.Units.largeSpacing
 
-        // Header area
-        RowLayout {
-            spacing: root.spacing
+        Layout.fillWidth: true
+        spacing: Kirigami.Units.smallSpacing
 
-            Kirigami.Icon {
-                id: icon
-                visible: source
-                implicitWidth: Kirigami.Units.iconSizes.large
-                implicitHeight: Kirigami.Units.iconSizes.large
-            }
+        Kirigami.Icon {
+            id: icon
+            visible: source
+            implicitWidth: Kirigami.Units.iconSizes.large
+            implicitHeight: Kirigami.Units.iconSizes.large
+        }
+
+        ColumnLayout {
+            Layout.fillWidth: true
+
+            spacing: Kirigami.Units.smallSpacing
 
-            ColumnLayout {
+            Kirigami.Heading {
+                id: titleHeading
                 Layout.fillWidth: true
+                level: 2
+                wrapMode: Text.Wrap
+                textFormat: Text.RichText
+                elide: Text.ElideRight
+            }
 
-                spacing: Kirigami.Units.smallSpacing
-
-                Kirigami.Heading {
-                    id: titleHeading
-                    Layout.fillWidth: true
-                    level: 2
-                    wrapMode: Text.Wrap
-                    textFormat: Text.RichText
-                    elide: Text.ElideRight
-                }
-
-                QQC2.Label {
-                    id: subtitleLabel
-                    Layout.fillWidth: true
-                    wrapMode: Text.Wrap
-                    elide: Text.ElideRight
-                    textFormat: Text.RichText
-                    visible: text.length > 0
-                }
+            QQC2.Label {
+                id: subtitleLabel
+                Layout.fillWidth: true
+                wrapMode: Text.Wrap
+                elide: Text.ElideRight
+                textFormat: Text.RichText
+                visible: text.length > 0
             }
         }
+    }
 
-        // Main content area, to be provided by the implementation
-        QQC2.Control {
-            id: contentsControl
+    // Main content area, to be provided by the implementation
+    QQC2.Control {
+        id: contentsControl
 
-            Layout.fillWidth: true
-            Layout.fillHeight: true
-            leftPadding: 0
-            rightPadding: 0
-            bottomPadding: 0
-            topPadding: 0
-        }
+        Layout.fillWidth: true
+        Layout.fillHeight: true
+
+        // make sure we don't add padding if there is no content
+        visible: contentItem !== null
+    }
 
-        // Footer area with buttons
-        QQC2.DialogButtonBox {
-            id: footerButtonBox
+    // Footer area with buttons
+    QQC2.DialogButtonBox {
+        id: footerButtonBox
 
-            Layout.fillWidth: true
-            leftPadding: 0
-            rightPadding: 0
-            bottomPadding: 0
-            topPadding: 0
+        Layout.fillWidth: true
 
-            visible: count > 0
+        visible: count > 0
 
-            onAccepted: root.window.accept()
-            onRejected: root.window.reject()
+        onAccepted: root.window.accept()
+        onRejected: root.window.reject()
 
-            Repeater {
-                model: root.actions
+        Repeater {
+            model: root.actions
 
-                delegate: QQC2.Button {
-                    action: modelData
-                }
+            delegate: QQC2.Button {
+                action: modelData
             }
         }
     }
-- 
GitLab

